### ISSUE TITLE
Encrypted send: Problems with placeholder filesystems

### ISSUE THREAD

**Issue by Ranlvor** [zrepl/zrepl#342]
When experimenting with encrypted sends on zrepl 0.3-rc1 I noticed that zrepl has problems when it needs to use placeholder filesystems on encrypted datasets.

Steps to reproduce:

- have zfs-filesystems on the source with native encryption: rpool/tmp/foo, rpool/tmp/foo/bar, rpool/tmp/foo/bar/baz
- have backups/zrepl/tadpole on remote server without any encryption keys loaded
- have backups/zrepl/tadpole on remote server configured as sink
- run zrepl with the following push-config:
```
    filesystems: {
            "rpool/tmp/foo": true,
            "rpool/tmp/foo/bar": false,
            "rpool/tmp/foo/bar/baz": true
    }
    send:
        encrypted: true
```
- replication fails with the following message:
```
         rpool/tmp/foo/bar/baz STEP-ERROR (step 0/1, 25.1 KiB/69.6 KiB)
            server error: zfs exited with error: exit status 1
            stderr: cannot create 'backups/zrepl/tadpole/rpool/tmp/foo/bar': encryption root's key is not loaded or provided
```

One possible workaround would be to run zfs create with -o encryption=off when creating placeholder filesystems on encrypted datasets.

```
# zfs create backups/zrepl/tadpole/rpool/tmp/test
cannot create 'backups/zrepl/tadpole/rpool/tmp/test': encryption root's key is not loaded or provided
```
but
```
# zfs create -o encryption=off backups/zrepl/tadpole/rpool/tmp/test
```
does work.

**Comment by problame** [comment]
I pushed a commit to my `problame/develop` branch that fixes your issue (see GitHub UI above)
This commit always sets `encryption=off` iff the parent filesystem is encrypted, and omits the option if no encryption is supported by the CLI or the parent is unencrypted.
Notably, the error message for 0.8.1 is different because the creation of unencrypted filesystems below encrypted ones is only allowed since 0.8.2. It's still an error message though, so I'm fine with no additional feature / version-compat detection there.

However, I am unsure whether this is actually what we want because overriding the `encryption` property in a placeholder will prevent scenarios where we receive an unencrypted stream into an encrypted dataset hierarchy with the key loaded ("encrypt-on-recv scenario").

@InsanePrawn this is somewhat related to your work in #285 .

The obvious answer is that we could add a which would control how placeholder filesystems should be created, e.g. just allow the user to specify a list of zfs create arguments and pass those down to the placeholder creation function.

But I feel like it's really not very obvious to the average user why this fails. So I would like zrepl to just "do the right thing" here.

https://github.com/zrepl/zrepl/blob/8e1937fe75c46393df6067b31715bb331c34ab44/config/config.go#L93-L99

Apparently I have already foreseen that we might want something like this in the future..

**Comment by problame** [comment]
@InsanePrawn and I discussed this out-of-band and came to the conclusion that, since there has not yet been a zrepl release that supports encryption, it is acceptable to not support the 'recv-unencrypted-into-encrypted_root_fs` use case in 0.3.

I'll update the commit message and let the commit trickle down to `master` before the nect `-rc`.

**Comment by problame** [comment]
fixed by 8ff83f2f1ac6faf364627f14c62880f35af4835e
